ml-proto/spec/ast.ml

```diff
@@ -20,10 +17,10 @@ and expr' =
   | Loop of expr list
   | Br of var * expr option
   | Br_if of var * expr option * expr
+  | Br_table of var list * var * expr option * expr
```

Comment:
For the binary encoding, I expect the default's `var` would go before the `var list` (that's what we've done in SM).  So since it's that way in both text and binary, perhaps the AST should match them so that the AST->binary mapping is not only regular w.r.t types but also order.